dask/dataframe/core.py

```diff
@@ -3295,9 +3302,18 @@ def pivot_table(self, index=None, columns=None,
         return pivot_table(self, index=index, columns=columns, values=values,
                            aggfunc=aggfunc)
 
-    def to_records(self, index=False):
+    def to_records(self, index=False, lengths=None):
         from .io import to_records
-        return to_records(self)
+
+        if lengths is True:
+            lengths = tuple(self.map_partitions(len).compute())
+
+        records = to_records(self)
+
+        chunks = self._validate_chunks(records, lengths)
+        records._chunks = chunks
```

Comment:
I'm not very familiar with record arrays, but apparently they're 1-dimensional?

```python
In [6]: df = pd.DataFrame({"A": [1, 2], "B": [3, 4]})

In [7]: df.to_records().ndim
Out[7]: 1
```

So this should (maybe) be `records._chunks = (chunks[0],)`.